src/Microsoft.ML.Data/Evaluators/Metrics/ConfusionMatrix.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data
+{
+    public sealed class ConfusionMatrix
+    {
+        public double[] PrecisionSums { get; }
+        public double[] RecallSums { get; }
+        public ReadOnlyMemory<char>[] Labels => PredictedLabelNames.ToArray(); //sefilipi: return as string[]
+
+        public double[][] ConfusionTableCounts { get; }
+
+        internal readonly List<ReadOnlyMemory<char>> PredictedLabelNames;
+        internal readonly bool Sampled;
+        internal readonly bool Binary;
+
+        private readonly IHost _host;
+
+        internal ConfusionMatrix(IHost host, double[] precisionSums, double[] recallSums, double[][] confusionTableCounts,
+            List<ReadOnlyMemory<char>> labelNames, bool sampled, bool binary)
+        {
+            _host = host;
+
+            PrecisionSums = precisionSums;
+            RecallSums = recallSums;
+            ConfusionTableCounts = confusionTableCounts;
+            PredictedLabelNames = labelNames;
+            Sampled = sampled;
+            Binary = binary;
+        }
+
+        public override string ToString() => MetricWriter.GetConfusionTableAsString(this, false);
+
+        public double GetCountForClassPair(string predictedLabel, string actualLabel)
+        {
+            int predictedLabelIndex = PredictedLabelNames.IndexOf(predictedLabel.AsMemory());
```

Comment:
>predictedLabel.AsMemory() [](start = 66, length = 25)

This API is not what we want anyway (since it relies on strings to index when there's no requirement that the strings *even exist* much less be unique), but in point of interest, please note that this code absolutely will not work.

```csharp
var m = "hello my friend".AsMemory();
var tokens = new List<ReadOnlyMemory<char>>();
tokens.Add(m.Slice(0, 5));
tokens.Add(m.Slice(6, 2));
tokens.Add(m.Slice(9, 6));

foreach (var token in tokens)
    Console.WriteLine($"'{token}'");

Console.WriteLine(tokens.IndexOf("friend".AsMemory()));
Console.WriteLine(tokens.IndexOf(tokens[2]));
```

If compiled and run on .NET core 2.1, you get this output.

```
'hello'
'my'
'friend'
-1
2
```

So, relying on `ReadOnlyMemory<char>`'s equality w.r.t. other similar items just is not a safe thing to do at all. You'll also recall all the bugs and barriers we had to fix back when we did the initial switch from our own `DvText` to `ReadOnlyMemory<char>`.